### DISCUSSION TITLE
Attiny 2313/4313 support

### DISCUSSION THREAD

**Discussion by NicoHood** [SpenceKonde/ATTinyCore#547] [Library discussion]
Hey!
I am the maintainer of the [Arduino PCINT library](https://github.com/NicoHood/PinChangeInterrupt). I got a request to also support the Attiny 2312/4313. I've seen you notes on that here:

https://github.com/SpenceKonde/ATTinyCore/blob/9371769d60ce64eaf7b96ff8a4a2bdc9394901f8/avr/extras/ATtiny_x313.md

In your [Attiny x313 variant definition](https://github.com/SpenceKonde/ATTinyCore/blob/59739fe353427fee590466a9ad6b54b7cd9b7742/avr/variants/tinyX313/pins_arduino.h#L5) you define `__AVR_ATtinyX313__` which I cannot find in the official avr-libc:
https://svn.savannah.gnu.org/viewvc/avr-libc/trunk/avr-libc/include/avr/io.h?view=markup

Now I am wondering how the board still works. I know that I can add support for the attinys by adding this definition as well, but I want to understand the whole process before adding it. ;-)

Here is the PR where I want to integrate this:
https://github.com/NicoHood/PinChangeInterrupt/pull/37

**Comment by SpenceKonde** [comment]
Oh - well, it is often very useful know if you are on a 84/44/24, or a 2313/4313, etc. So for convenience in the variants I have a #define that duplicates the format of the avr-libc part identification ones, only witth the part of the part number which defines the size replaced with an X.  So `defined(__AVR_ATtinyX313__)` is syntactic sugar for `(defined(__AVR_ATtiny2313__) || defined(__AVR_ATtiny4313__))`. They're used often internally within the core  too.
This in no way interferes with the avr-libc supplied #defines - since it sounds like in your case the two are different, it would make sense to just test the standard defines, which - the avr-libc or io headers or wherever exactly that comes from supplies those #defines, you should be able top use them and ignore the `__AVR_ATtinyX313__` define.

**Reply by SpenceKonde** [reply]
Although I'm one of those purists who doesn't like the idea of a library for PCINTs. If f there's anything I could do to makeprocess of adding support for the tinies to this library, just let me know. :-)

**Reply by NicoHood** [reply]
The guy who was requesting the feature will give it a test tomorrow. If that does not work, I'd come back to you, as I have no real board available. Thanks a lot so far!

**Reply by SpenceKonde** [reply]
Uh - it looks like the coide in the repo is still testing for `__AVR_ATtinyX313__`  Why bother with that, if already testing for `__AVR_ATtiny4313__`  and `__AVR_ATtiny2313A__` and `__AVR_ATtiny2313__` ?

Looking at that PR, it looks like all that needs to get added is a PCINT_INPUT_PORTn PORTx macro for each bank of PCINTs.

**Reply by SpenceKonde** [reply]
after I tie up some loose ends on the modern cores hopefully today. I could make a list of them for the tinyAVRs you don't have yet. for the sake of completeness.

**Reply by NicoHood** [reply]
I've added that definition, because it was not working before. It is just a blind guess, as I have no boards for testing.
